src/components/ProductsTable/ProductsTable.tsx

```diff
@@ -1,28 +1,54 @@
-import './ProductsTable.scss';
-import { ProductsTableRow } from 'components/ProductsTableRow/ProductsTableRow';
+import { ProductsTableRow } from './ProductsTableRow';
 import { useProducts } from 'hooks/useProducts';
 import { ReactElement } from 'react';
+import { useSearch } from 'components/Search/SearchContext';
+import { Search } from 'components/Search/Search';
+import { Button } from 'components/Buttons/Button';
+import { useProductFormContext } from 'components/ProductForm/ProductFormContext';
+import { ProductDto } from 'dtos/ProductDto';
+import { LoaderOverlay } from 'components/LoaderOverlay/LoaderOverlay';
 
 export const ProductsTable = (): ReactElement => {
   const { products } = useProducts();
+  const formatting = ' px-4 py-2 font-normal ';
```

Comment:
is this necessary? Might be wrong, but if you add this to thead and tbody, it should apply to all children. No need to apply it everywhere and pass it down

Reply:
it has to be applied individually, text-bold is being automatically applied to each `<th>` component so it has to be directly put in each element else be overrided, alongside the padding which isn't inherited.

Reply:
If you add it to the <tr> it should work fine. When you're adding the same classes like this to many things it's usually an indication it's the wrong approach

Reply:
I cannot get any combination margins, padding, gap or space to work on any of the html elements
I based the table [on these](https://flowbite.com/docs/components/tables/) which have padding for every element

Reply:
yeah styling of `thead` and `tbody` is very limited, you cannot add stuff like padding to them so you have to add it to the individual cells instead. Combine that with tailwind and you get duplicate code 😬

Reply:
I reckon in situations like these it's probably fine to have duplicate classes, we are using tailwind after all and its better to just have all of the classes visible than having half of them abstracted away in a variable somewhere

Reply:
sorry I meant add it to the table row <tr>, if that doesn't work then leave it

Reply:
@hazel-davis @mbaltrimas